src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -2136,4 +2141,104 @@ protected function appendOrderArgumentsToUrl($order_id, $order_key, $webhook_url
+     *
+     * @param $statusTypeString
+     *
+     * @return mixed|void|null
+     */
+    protected function overwriteFilters($new_order_status, $statusTypeString)
+    {
+        // Overwrite plugin-wide
+        $new_order_status = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString,
+            $new_order_status
+        );
+
+        // Overwrite gateway-wide
+        $new_order_status = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString . $this->id,
+            $new_order_status
+        );
+        return $new_order_status;
+    }
+
+    /**
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     *
+     * @return bool
+     */
+    protected function isPartialRefund($payment)
+    {
+        return ($payment->amount->value - $payment->amountRefunded->value) != 0;
+    }
+
+    /**
+     * @param WC_Order                                                $order
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     */
+    protected function processUpdateStateRefund(WC_Order $order, $payment)
+    {
+        if ($this->isPartialRefund($payment)) {
+            $this->updateStateRefund(
+                $order,
+                $payment,
+                self::STATUS_ON_HOLD,
+                '_order_status_partially_refunded',
+                'partially refunded'
+            );
+            return;
+        }
+        $this->updateStateRefund(
+            $order,
+            $payment,
+            self::STATUS_REFUNDED,
+            '_order_status_refunded',
+            'refunded'
+        );
+    }
+
+    /**
+     * @param WC_Order                                                $order
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     * @param                                                         $new_order_status
+     * @param                                                         $statusFilterString
+     * @param                                                         $statusNoteString
+     */
+    protected function updateStateRefund(
+        WC_Order $order,
+        $payment,
+        $new_order_status,
+        $statusFilterString,
+        $statusNoteString
+    ) {
+        // New order status
+        $new_order_status = $this->overwriteFilters(
+            $new_order_status,
+            $statusFilterString
+        );
+        $payment_method_title = $this->getPaymentMethodTitle($payment);
+
+        // Update order status for order with charged_back payment, don't restore stock
+        $this->updateOrderStatus(
+            $order,
+            $new_order_status,
+            sprintf(
+            /* translators: Placeholder 1: payment method title, placeholder 2: payment ID */
+                __(
+                    '%s payment %s via Mollie (%s). You will need to manually review the payment (and adjust product stocks if you use it).',
```

Comment:
When there are more parameters it's better to use positional. `%$1s`, `%$2s` and so on...

Reply:
Strange, I was trying the positional https://3v4l.org/BJ5k4 and with printf it works %1$s, but with sprintf it does not tried %$1s and %1$s and gives error...any clues?

Reply:
Because you're using double quotes `echo sprintf("[%1s %2s]",$str1, $str2);` and php think it's a reference to a variable `$s`. you have to use single quotes. `echo sprintf('[%1$s %2$s]',$str1, $str2);`

See https://3v4l.org/04pIG